src/icalendar/tests/test_issue_722_generate_vtimezone.py

```diff
@@ -0,0 +1,408 @@
+        ("Europe/Berlin", datetime(2024, 10, 27, 2, 59, 59), "CEST"),
+
+        ("Europe/Berlin", datetime(2024, 10, 27, 3, 0), "CET"),
+        ("Europe/Berlin", datetime(2024, 10, 27, 3, 0, 1), "CET"),
+        ("Europe/Berlin", datetime(2024, 10, 27, 4, 0), "CET"),
+
+        # transition times from https://www.zeitverschiebung.net/de/timezone/america--new_york
+        ("America/New_York", datetime(1970, 1, 1), "EST"),
+        # Daylight Saving Time
+        ("America/New_York", datetime(2024, 11, 3, 0, 0), "EDT"),
+        ("America/New_York", datetime(2024, 11, 3, 1, 0), "EDT"),
+        ("America/New_York", datetime(2024, 11, 3, 1, 59, 59), "EDT"),
+        # 03.11.2024 2:00am -> 1:00am Standard
+        # ("America/New_York", datetime(2024, 11, 3, 2, 0), "EDT"),
+        ("America/New_York", datetime(2024, 11, 3, 2, 0, 1), "EST"),
+        ("America/New_York", datetime(2024, 11, 3, 3, 0), "EST"),
+        ("America/New_York", datetime(2025, 3, 9, 1, 0), "EST"),
+        ("America/New_York", datetime(2025, 3, 9, 1, 59, 59), "EST"),
+        ("America/New_York", datetime(2025, 3, 9, 2, 0), "EST"),
+        # 09.03.2025 2:00am -> 3:00am Daylight Saving Time
+        ("America/New_York", datetime(2025, 3, 9, 3, 0), "EDT"),
+        ("America/New_York", datetime(2025, 3, 9, 3, 1, 1), "EDT"),
+        ("America/New_York", datetime(2025, 3, 9, 4, 0), "EDT"),
+    ]
+)
+def test_check_datetimes_around_transition_times(tzp, tzid, dt, tzname):
+    """We should make sure than the datetimes with the generated timezones
+    work as expected: They have the right UTC offset, dst and tzname.
+    """
+    message = f"{tzid}: {dt} (expected in {tzname})"
+    expected_dt = tzp.localize(dt, tzid)
+    component = Timezone.from_tzinfo(tzp.timezone(tzid))
+    generated_tzinfo = component.to_tz(tzp, lookup_tzid=False)
+    generated_dt = dt.replace(tzinfo=generated_tzinfo)
+    print(generated_tzinfo)
+    if tzp.uses_pytz():
+        # generated_dt = generated_tzinfo.localize(dt)
+        generated_dt = generated_tzinfo.normalize(generated_dt)
+        if dt in (
+                datetime(2024, 10, 27, 1, 0),
+                datetime(2024, 11, 3, 1, 59, 59),
+                datetime(2024, 11, 3, 1, 0),
+                datetime(2024, 11, 3, 0, 0),
+                datetime(2024, 10, 27, 2, 59, 59),
+                datetime(2024, 10, 27, 2, 30),
+                datetime(2024, 10, 27, 2, 0),
+                datetime(2024, 10, 27, 1, 0)
+            ):
+            pytest.skip("We do not know how to do this. PYTZ-TODO")
+    assert generated_dt.tzname() == expected_dt.tzname() == tzname, message
+    assert generated_dt.dst() == expected_dt.dst(), message
+    assert generated_dt.utcoffset() == expected_dt.utcoffset(), message
+
+
+@pytest.mark.parametrize(
+    "uid", [0, 1, 2, 3]
+)
+def test_dateutil_timezone_when_time_is_going_backwards(calendars, tzp, uid):
+    """Sometimes, times can be ambiguous: We go back from 3am to 2 am.
```

Comment:
Although those who work with datetime and DST shifts have the experience of being bitten by this, we should explain it in detail. The explanation might also be helpful in the narrative docs, but I'm not sure where. There are also some really good PyCon videos and talks that discuss working with datetime.

```suggestion
    """When going from Daylight Saving Time to Standard Time, times can be ambiguous.
    For example, at 3:00 AM, the time falls back to 2:00 AM, repeating a full hour of times from 2:00 AM to 3:00 AM on the same date.
```